pipeline/authz/opa_engine.go

```diff
@@ -0,0 +1,163 @@
+package authz
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/asaskevich/govalidator"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/oathkeeper/helper"
+	"github.com/ory/oathkeeper/pipeline"
+	"github.com/ory/oathkeeper/pipeline/authn"
+
+	"github.com/ory/x/httpx"
+)
+
+type AuthorizerOPAEngineConfiguration struct {
+	RequiredAction   string `json:"required_action" valid:",required"`
```

Comment:
This assumes the input parameters for OPA to have a specific format. That is an antipattern because it is specific to your use case. Instead, we should use go templates to generate the input values, as it is done with the [header](https://github.com/ory/oathkeeper/blob/master/pipeline/mutate/mutator_header.go) mutator.